add cron example

KuthumiPepple opened the pull request:
@delawen Cron example runs successfully.

Delawen commented:
Can you rename the folder to `13-cron`?

Also, can you add a bit more context on what the route does? Like here https://github.com/apache/camel-k-examples/tree/main/12-aws-secret-manager-vault At least add the command line to run it.

KuthumiPepple commented:
> Can you rename the folder to `13-cron`?
> 
> Also, can you add a bit more context on what the route does? Like here https://github.com/apache/camel-k-examples/tree/main/12-aws-secret-manager-vault At least add the command line to run it.

I've renamed and added a didact tutorial. Let me know if more changes are needed.

squakez reviewed (commented):
I think Didact is going to be discontinued. Its repo was archived not long time ago: https://github.com/redhat-developer/vscode-didact

squakez commented:
I don't like having number in front of the directory either. I prefer having an index as we used to do in the Camel K examples: https://github.com/apache/camel-k/tree/main/examples

KuthumiPepple commented:
> I think Didact is going to be discontinued. Its repo was archived not long time ago: https://github.com/redhat-developer/vscode-didact

Hmm. I didn't know that. Makes sense why new examples didn't use it

KuthumiPepple commented:
> I don't like having number in front of the directory either. I prefer having an index as we used to do in the Camel K examples: https://github.com/apache/camel-k/tree/main/examples

I like the index too. I'll speak to my mentor about it.

Delawen commented:
Whatever @squakez says, we do :)

squakez commented:
> Whatever @squakez says, we do :)
> 
> (except re-writing the whole Camel K in Javascript, there must always be red lines!)

I am flattered :) However, my opinion counts as anybody's one else. I may be wrong so we better have open discussion and submit to vote when no agreement is reached.

KuthumiPepple commented:
@squakez what's the difference between custom and generic examples? In the root folder's readme, I see examples grouped under these categories.

oscerd commented:
I think it makes sense to remove them. We should have just generic and kamelets.

KuthumiPepple commented:
Cleaned up commit history. Improved cron example in generic-examples directory.

apupier commented on generic-examples/cron/README.md:
```diff
@@ -1,3 +1,105 @@
+The following requirements are optional. They don't prevent the execution of the demo, but may make it easier to follow.
+
+**VS Code Extension Pack for Apache Camel**
+The VS Code Extension Pack for Apache Camel provides a collection of useful tools for Apache Camel K developers,
```
```suggestion
The [VS Code Extension Pack for Apache Camel](https://marketplace.visualstudio.com/items?itemName=redhat.apache-camel-extension-pack) provides a collection of useful tools for Apache Camel K developers,
```

KuthumiPepple replied:
Done. Thanks

KuthumiPepple commented:
@tadayosi I think this one can be merged too
